core/src/main/java/ai/timefold/solver/core/impl/move/streams/dataset/DataStreamFactory.java

```diff
@@ -10,24 +9,23 @@
 
 import ai.timefold.solver.core.api.score.stream.Joiners;
 import ai.timefold.solver.core.impl.domain.solution.descriptor.SolutionDescriptor;
-import ai.timefold.solver.core.impl.move.streams.maybeapi.stream.DataStreamFactory;
-import ai.timefold.solver.core.impl.move.streams.maybeapi.stream.SolutionExtractor;
+import ai.timefold.solver.core.impl.move.streams.FromSolutionValueCollectingFunction;
 import ai.timefold.solver.core.impl.move.streams.maybeapi.stream.UniDataStream;
 
-import org.jspecify.annotations.NonNull;
+import org.jspecify.annotations.NullMarked;
 
-public final class DefaultDataStreamFactory<Solution_> implements DataStreamFactory<Solution_> {
+@NullMarked
+public final class DataStreamFactory<Solution_> {
```

Comment:
Should this class be abstract, or will there be no default implementation?

Reply:
At the moment, I'm only doing things I need. If I ever need a public and private version of this, I'll change it - right now, it looks as if it will be enough if this is completely in impl, and none of it in API. Therefore no need to create abstractions.